ChessBoardApp/ChessBoard/InputValidation.cs

```diff
@@ -0,0 +1,42 @@
+using Serilog;
+
+namespace ChessBoardApp.ChessBoard
+{
+
+
+    public class InputValidation
+    {
+        private readonly ILogger _logger;
+
+        public InputValidation(ILogger logger)
+        {
+            _logger = logger;
+        }
+        public class ValidationResult
+        {
+            public bool IsValid { get; set; }
+            public string ErrorMessage { get; set; }
+
+            public ValidationResult(bool isValid, string errorMessage)
+            {
+                IsValid = isValid;
+                ErrorMessage = errorMessage;
+            }
+        }
+
+
+        public ValidationResult ValidateInput(string input)
+        {
+            bool isValid = int.TryParse(input, out var validInteger) && validInteger > 0;
+
+            if (!isValid)
+            {
+                string errorMessage = "Invalid input. An integer higher than 0 is required";
+                _logger.Error($"Invalid input '{input}'");
+                return new ValidationResult(false, errorMessage);
+            }
+
+            return new ValidationResult(true, null);
```

Comment:
No need to adjust, just an insight - we can use static methods as constructors for validation result, then we  would not need to think what "true, null" means when reading the code. 
To do this you can change ctor for `ValidationResult` to private and add Success/Failure methods:
![image](https://github.com/Pauliukas7/dotnet-console-app/assets/2812503/ef191105-0f64-4997-9d2d-a797508261ad)

Reply:
I liked the idea, so I adjusted it. As you've mentioned the ValidateInput method now only returns predefined objects - looks cleaner